### ISSUE TITLE
Project Templates

### ISSUE THREAD

**Issue by throughnothing** [purescript-contrib/pulp#271]
It would be nice to be able to develop project templates/skeletons so that library / app authors could provide "quick start" templates for people looking to try things out.  It would also be useful for people who commonly start repos with custom setups/libs that they frequently want.  This same idea had been mentioned [here](https://github.com/bodil/pulp/issues/78#issuecomment-151242685).

**Comment by paf31** [comment]
I think it would be great to allow packages to define their own templates. So for example, Thermite could include a template directory with a starter project, and then the user could simply type `pulp init thermite`. Pulp would simply copy the template directory into the current directory and initialize `bower.json` with only `purescript-thermite`.

**Comment by throughnothing** [comment]
@paf31 Yes, I like that idea, but I also think you could point to a template at a git/http resource.  That way pulp itself doesn't need to be full of all kinds of templates, but it has some template format that can be pointed to for easy setup.  I think pointing to a pulp-template-repo or gist, or file over HTTP is the simplest for pulp, and for library developers to provide as a quick start.

**Comment by epicallan** [comment]
@hdgarrood how do you feel about a pull request with @throughnothing implementation.

**Comment by hdgarrood** [comment]
Sure, I'll have a look. I have an exam later this week (my last one of the semester, yay) so I can't promise I'll be able to review quickly.

**Comment by hdgarrood** [comment]
I'm not sure I understand this proposal. What does this give you over asking users to run e.g. `git clone https://github.com/user/purescript-thermite-template`?
